client/signup/site-mockup/index.jsx

```diff
@@ -49,6 +49,18 @@ class SiteMockups extends Component {
 		this.state = this.getNewFontLoaderState( props );
 	}
 
+	shouldComponentUpdate( nextProps ) {
+		// Debouncing updates to the preview content
+		// prevents the flashing effect.
+		if ( nextProps.verticalPreviewContent !== this.props.verticalPreviewContent ) {
+			this.updateDebounced();
+			return false;
+		}
+
+		return true;
+	}
+	updateDebounced = debounce( this.forceUpdate, 777, { leading: false } );
```

Comment:
Just an FYI that `leading` is defaulted to `false`: https://lodash.com/docs/4.17.11#debounce :)